src/WorkflowManager/Authentication/Extensions/MonaiAuthenticationExtensions.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+ * Copyright 2022 MONAI Consortium
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Monai.Deploy.WorkflowManager.Common.Extensions;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authentication;
+using Monai.Deploy.WorkflowManager.Authentication.Middleware;
+
+namespace Monai.Deploy.WorkflowManager.Authentication.Extensions
+{
+    public static class MonaiAuthenticationExtensions
+    {
+        /// <summary>
+        /// Adds MONAI OpenID Configuration to services.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+        /// <param name="configuration">The <see cref="IConfiguration"/>.</param>
+        public static IServiceCollection AddMonaiAuthentication(
+            this IServiceCollection services,
+            IConfiguration configuration)
+        {
+            if (configuration.BypassAuth())
+            {
+                services.AddAuthentication(options => options.DefaultAuthenticateScheme = "testing")
```

Comment:
Should we refactor all strings into a resource file?

Reply:
Think I got all strings. moved them into AuthKeys.cs